ocdata/core/surface.py

```diff
@@ -0,0 +1,275 @@
+import math
+import os
+import pickle
+from collections import defaultdict
+
+import numpy as np
+from ase import neighborlist
+from ase.constraints import FixAtoms
+from pymatgen.analysis.local_env import VoronoiNN
+from pymatgen.core import Composition
+from pymatgen.io.ase import AseAtomsAdaptor
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from ocdata.configs.constants import MIN_XY
+from ocdata.core import Bulk
+
+
+class Surface:
+    """
+    Initializes a surface object, i.e. a particular slab tiled along xyz, in
+    one of 2 ways:
+    - Pass in a Bulk object and a slab 4-tuple containing
+    (atoms, miller, shift, top).
+    - Pass in a Bulk object and randomly sample a surface.
+
+    Arguments
+    ---------
+    bulk: Bulk
+        Corresponding bulk object.
+    slab: tuple
+        4-tuple containing (atoms, miller, shift, top).
+    """
+
+    def __init__(
+        self,
+        bulk: Bulk,
+        slab: tuple = None,
```

Comment:
After removing the global constant, let's feed in `min_xy` as an argument and just set it to default to whatever MIN_XY was.